libtcmu.c

```diff
@@ -618,7 +797,19 @@ static void remove_device(struct tcmulib_context *ctx,
 
 	darray_remove(ctx->devices, i);
 
+        struct tcmu_aio_queue *aio_wq = &dev->work_queue;
+
+        /*
+         * The order of cleaning up worker threads and calling ->removed()
+         * is important: for sync handlers, the worker thread needs to be
+         * terminated before removing the handler (i.e., calling handlers
+         * ->close() callout) in order to ensure that no store callouts
+         * are getting invoked when shutting down the handler.
+         */
+        cancel_thread(aio_wq->aio_wq_thread);
         dev->handler->removed(dev);
```

Comment:
What happens to executing commands that are running when cancel_thread is called? If it is on the track_queue will it ever get completed andcleaned up?

Is the problem for this case that we do not know what command processing state the tcmur_handler was in when the cancel was called?